src/__tests__/GameRequest.test.js

```diff
@@ -1,21 +1,116 @@
+            expect(component.getFormattedTimeDiff(Date.now()-(24 * 60 * 60000))).toBe('1 day ago');
+        });
+    });
+    describe('render', () => {
+        test('should render component using quiplash data', async () => {
+            let {container} = render(<GameRequest {...propsQuip} />);
+            try {
+                await waitFor(() => screen.getByText(/Quiplash/i));
+                const element = screen.getByText(/Quiplash/i);
+                expect(element).toBeInTheDocument();
+            } catch (e) {
+                throw e;
             }
-        };
-        render(<GameRequest {...props} />);
-        await waitFor(() => screen.getByText(/Quiplash/i));
-        const element = screen.getByText(/Quiplash/i);
-        expect(element).toBeInTheDocument();
+            expect(container).toMatchSnapshot();
+        });
+        test('should render component using survive the internet data', () => {
+            const shallowRenderer = createRenderer();
+            shallowRenderer.render(<GameRequest {...propsSTI} />);
+            let instance = shallowRenderer.getMountedInstance();
+            instance.setState({
+                activityStatus: ActivityStatus.ACTIVE,
+                timeDiff: '2 minutes ago'
+            });
+            expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
+            instance.setState({
+                activityStatus: ActivityStatus.IDLE,
+                timeDiff: '3 minutes ago'
+            });
+            expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
+            instance.setState({
+                activityStatus: ActivityStatus.DISCONNECTED,
+                timeDiff: '4 minutes ago'
+            });
+            expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
+            shallowRenderer.unmount();
+        });
+        test('should render component using survive the internet data and updated status info', async () => {
+            let props = Object.assign({}, propsSTI, {
+                getActivity: jest.fn()
+                    .mockResolvedValueOnce(3)
+                    .mockResolvedValueOnce(2)
+                    .mockResolvedValueOnce(1),
+                onDelete: jest.fn(),
+                toggleLock: jest.fn()
+            })
+            const shallowRenderer = createRenderer();
+            shallowRenderer.render(<GameRequest {...props} />);
+            let instance = shallowRenderer.getMountedInstance();
+            instance.setState({
+                activityStatus: ActivityStatus.IDLE,
+                timeDiff: '2 minutes ago'
+            });
+            let component = shallowRenderer.getRenderOutput();
+            await component.props.children[1].props.onMouseEnter();
+            expect(props.getActivity).toHaveBeenCalledTimes(1);
+
+            let optionsChildren = component.props.children[1].props.children.props.children[1].props.children;
```

Comment:
1. yeahhhh this is kind of a mood
2. can both present-me and future-me at least get a comment describing this scuba diving expedition, so it's not quite so opaque if/when we change something later? because I am not parsing this at all right now lol

Reply:
Oh, duh, I'll a comment about what's going on.  Essentially, it's testing the functions attached to the component and verifying they trigger the expected functions. The expedition is just a way to grab all of the children inside div.options